src/sentry/api/event_search.py

```diff
@@ -1480,18 +1516,22 @@ def __init__(
         :param str transform: NOTE: Use aggregate over transform whenever possible.
             An aggregate string to be passed to snuba once formatted. The arguments
             will be filled into the string using `.format(...)`.
-        :param str result_type: The resulting type of this function. Can be any of the following
-            (duration, string, number, integer, percentage, date).
+         :param str result_type_fn: A function to call with in order to determine the result type.
+            This function will be passed the list of argument classes and argument values. This should
+            be tried first as the source of truth if available.
+        :param str default_result_type: The default resulting type of this function. Can be any of the
+            following (duration, string, number, integer, percentage, date).
```

Comment:
Could we move these types to a constant, that way we can assert that the type of a constructed function is within the constant? As well as using the constant to define the functions